test/kinetics/kineticsFromYaml.cpp

```diff
@@ -91,6 +91,119 @@ TEST(Reaction, ThreeBodyFromYaml2)
 
     const auto& rate = std::dynamic_pointer_cast<ArrheniusRate>(R->rate());
     EXPECT_DOUBLE_EQ(rate->preExponentialFactor(), 1.2e11);
+
+    AnyMap input = R->parameters();
+    EXPECT_FALSE(input.hasKey("type"));
+    EXPECT_TRUE(input.hasKey("efficiencies"));
+
+    auto efficiencies = input["efficiencies"].asMap<double>();
+    EXPECT_EQ(efficiencies.size(), 2);
+    EXPECT_EQ(efficiencies["AR"], 0.83);
+    EXPECT_EQ(efficiencies["H2O"], 5.);
+}
+
+TEST(Reaction, ThreeBodyFromYaml3)
+{
+    auto sol = newSolution("gri30.yaml", "", "None");
+    AnyMap rxn = AnyMap::fromYamlString(
+        "{equation: CH2 + M <=> CH2(S) + M,"
+        " rate-constant: {A: 5.0e+9, b: 0.0, Ea: 0.0}}");
+
+    auto R = new Reaction(rxn, *(sol->kinetics()));
+    EXPECT_EQ(R->type(), "three-body-Arrhenius");
+    EXPECT_TRUE(R->usesThirdBody());
+    EXPECT_EQ(R->thirdBody()->name(), "M");
+
+    const auto& rate = std::dynamic_pointer_cast<ArrheniusRate>(R->rate());
+    EXPECT_DOUBLE_EQ(rate->preExponentialFactor(), 5.0e+9);
+}
+
+TEST(Reaction, ThreeBodyFromYaml4)
+{
+    auto sol = newSolution("gri30.yaml", "", "None");
+    AnyMap rxn = AnyMap::fromYamlString(
+        "{equation: CH2 + O2 <=> CH2(S) + O2,"
+        " type: three-body,"
+        " rate-constant: {A: 5.0e+9, b: 0.0, Ea: 0.0}}");
+
+    auto R = newReaction(rxn, *(sol->kinetics()));
+    EXPECT_EQ(R->type(), "three-body-Arrhenius");
+    EXPECT_TRUE(R->usesThirdBody());
+    EXPECT_EQ(R->thirdBody()->name(), "O2");
+
+    const auto& rate = std::dynamic_pointer_cast<ArrheniusRate>(R->rate());
+    EXPECT_DOUBLE_EQ(rate->preExponentialFactor(), 5.0e+9);
+
+    AnyMap input = R->parameters();
+    EXPECT_EQ(input.getString("type", ""), "three-body");
+    EXPECT_FALSE(input.hasKey("efficiencies"));
+    EXPECT_FALSE(input.hasKey("default-efficiency"));
+}
+
+TEST(Reaction, ThreeBodyFromYaml5)
+{
+    auto sol = newSolution("gri30.yaml", "", "None");
+    AnyMap rxn = AnyMap::fromYamlString(
+        "{equation: CH2 + O2 <=> CH2 + O2,"
+        " type: three-body,"
+        " rate-constant: {A: 5.0e+9, b: 0.0, Ea: 0.0}}");
+
+    EXPECT_THROW(newReaction(rxn, *(sol->kinetics())), CanteraError);
+}
+
+TEST(Reaction, ThreeBodyFromYaml6)
+{
+    auto sol = newSolution("gri30.yaml", "", "None");
+    AnyMap rxn = AnyMap::fromYamlString(
+        "{equation: CH2 + O2 <=> CH2 + O2,"
+        " rate-constant: {A: 5.0e+9, b: 0.0, Ea: 0.0},"
+        " default-efficiency: 0.,"
+        " efficiencies: {O2: 1.}}");
+
+    auto R = newReaction(rxn, *(sol->kinetics()));
+    EXPECT_EQ(R->type(), "three-body-Arrhenius");
+    EXPECT_TRUE(R->usesThirdBody());
+    EXPECT_EQ(R->thirdBody()->name(), "O2");
+
+    const auto& rate = std::dynamic_pointer_cast<ArrheniusRate>(R->rate());
+    EXPECT_DOUBLE_EQ(rate->preExponentialFactor(), 5.0e+9);
+
+    AnyMap input = R->parameters();
```

Comment:
I think these tests should be run with `false` passed to the `parameters()` method to suppress the keys that came directly from the input, and test that the method is actually recreating the structure.